gateway/handle_observability.go

```diff
@@ -18,14 +20,13 @@ func (gw *Handle) NewSourceStat(arctx *gwtypes.AuthRequestContext, reqType strin
 	}
 }
 
-func (gw *Handle) newSourceStatTagsWithReason(arctx *gwtypes.AuthRequestContext, reqType, reason string) stats.Tags {
+func (gw *Handle) newSourceStatTagsWithReason(s backendconfig.SourceT, reqType, reason string) stats.Tags {
 	tags := stats.Tags{
-		"source":       arctx.SourceTag(),
-		"source_id":    arctx.SourceID,
-		"write_key":    arctx.WriteKey,
+		"source":       misc.GetTagName(s.WriteKey, s.Name),
+		"source_id":    s.SourceDefinition.ID,
```

Comment:
Has something changed..?
```suggestion
		"source_id":    s.ID,
```